java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSubscriptionImpl.java

```diff
@@ -328,34 +236,19 @@ private boolean tryRecordDisconnect() {
         return CONNECTED_UPDATER.compareAndSet(this, 1, 0);
     }
 
-    private void trySignalCompletion() {
-        trySignalCompletion(null);
-    }
-
-    private void trySignalCompletion(@Nullable final Throwable t) {
-        if (!COMPLETED_UPDATER.compareAndSet(BarrageSubscriptionImpl.this, 0, 1)) {
-            return;
+    private boolean onFutureCancel(boolean cancelResult) {
```

Comment:
It seems to me like it would be better to get rid of the argument and only call this if the argument would have been `true`. That way it matches `onFutureComplete()`, and we only have one pattern.